### PR TITLE
Controls for conf and channel

### PR DESCRIPTION
**Pull request by jscasca**
Weird that I could not create a fn template to copy the beforeMount fn
so I had to copy for each story. I will revisit this in the future and
try to come uo with something

### CODE REVIEW

**Comment by jscasca** [comment]
> why?
You ask regarding my lack of fn template. I don't know, the component wouldn't load. Weird....

File: src/stories/Editor.stories.tsx [hunk 1]
```diff
@@ -11,53 +13,130 @@ const content = `
   <strong><span style="font-size: 14pt;"><span style="color: #7e8c8d; font-weight: 600;">No matter what you're building, TinyMCE has got you covered.</span></span></strong>
 </p>`;
 
+let lastChannel = '5';
+const getConf = (stringConf) => {
+  let conf  = {};
+  console.log('parsing: ', stringConf);
+  try {
+    conf = JSON.parse(stringConf);
```
Comment (tiny-james) [inline src/stories/Editor.stories.tsx:21]: Will JSON be enough? That won't work for functions.

**Review by tiny-james — APPROVED** [review]

File: src/stories/Editor.stories.tsx (hunk 1 above)
Comment (jscasca) [reply src/stories/Editor.stories.tsx:21]: Do we to have functions for testing purposes only? We can always change it to eval instead

File: src/stories/Editor.stories.tsx [hunk 2]
```diff
@@ -11,53 +13,130 @@ const content = `
...
   },
-  template: '<div><editor :api-key="apiKey" :initialValue="content" /></div>'
+  template: '<div ><p>Ready</p><editor :api-key="apiKey" :initialValue="content" :cloud-channel="cloudChannel" :init="conf" /></div>'
 });
 
-export const Inline: Story = () => ({
+export const Inline: Story = (args) => ({
   components: { Editor },
-  data: () => ({ content }),
+  setup() {
```
[the remark this answers is not in the document]
Comment (jscasca) [reply src/stories/Editor.stories.tsx:77]: The `(args) => ({components, setup, template})` was the one that I couldn't copy. I tried making a fn `(template) => (args) => {}` and it didn't work.

File: src/stories/Editor.stories.tsx [hunk 3]
```diff
@@ -11,53 +13,130 @@ const content = `
...
+}
+
+
 export default {
   title: 'Editor',
   component: Editor,
+  argTypes: {
+    channel: {
+      defaultValue: '5',
+      options: ['5', '5-dev', '6-dev', '6-testing', '6-stable'],
```
Comment (lnewson) [inline src/stories/Editor.stories.tsx:35]: Should this include `5-testing`?

File: src/stories/Editor.stories.tsx [hunk 4]
```diff
@@ -11,53 +13,130 @@ const content = `
...
   }
 };
 
-export const Iframe: Story = () => ({
+export const Iframe: Story = (args) => ({
   components: {Editor},
   setup() {
+    onBeforeMount(() => {
+      if (args.channel !== lastChannel) {
+        delete (window as any).tinymce;
```
Comment (lnewson) [inline src/stories/Editor.stories.tsx:55]: You'll likely also want to delete `window.tinyMCE` if you want to clean up all references.

File: src/stories/Editor.stories.tsx [hunk 5]
```diff
@@ -11,53 +13,130 @@ const content = `
...
+        ScriptLoader.reinitialize();
+        ScriptLoader.load(document, `https://cdn.tiny.cloud/1/${apiKey}/tinymce/${args.channel}/tinymce.min.js`, () => {
+          console.log('script ready');
+        });
+        lastChannel = args.channel;
+      }
+    });
+    const cc = args.channel || lastChannel;
+    const conf = getConf(args.conf);
+    const log = (e, editor) => {console.log(e);};
```
Comment (lnewson) [inline src/stories/Editor.stories.tsx:123]: Does this need types?

File: src/stories/Editor.stories.tsx [hunk 6]
```diff
@@ -68,12 +147,32 @@ export const Controlled: Story = () => ({
...
+        ScriptLoader.load(document, `https://cdn.tiny.cloud/1/${apiKey}/tinymce/${args.channel}/tinymce.min.js`, () => {
+          console.log('script ready');
+        });
+        lastChannel = args.channel;
+      }
+    });
+    const cc = args.channel || lastChannel;
+    const conf = getConf(args.conf);
+    const disabled = ref(false);
+    const toggleDisabled = (_e) => {
```
Comment (lnewson) [inline src/stories/Editor.stories.tsx:166]: Types for `_e`?

**Review by lnewson — APPROVED** [review]

File: src/stories/Editor.stories.tsx (hunk 5 above)
Comment (jscasca) [reply src/stories/Editor.stories.tsx]: Added types

File: src/stories/Editor.stories.tsx (hunk 4 above)
Comment (jscasca) [reply src/stories/Editor.stories.tsx]: Deleting this as well

File: src/stories/Editor.stories.tsx (hunk 3 above)
Comment (jscasca) [reply src/stories/Editor.stories.tsx]: Added '5-testing'

File: src/stories/Editor.stories.tsx (hunk 1 above)
Comment (jscasca) [reply src/stories/Editor.stories.tsx]: Move to use `new Function()` instead

**Pull request merged**